cpp/modmesh/buffer/SimpleArray.hpp

```diff
@@ -84,6 +84,7 @@ namespace detail
 
 using shape_type = small_vector<size_t>;
 using sshape_type = small_vector<ssize_t>;
+using slice_type = small_vector<int>;
```

Comment:
While moved it here, it becomes obvious that `int` does not look right.  `slice_type` should be interchangeable with `sshape_type`.  Please review and make them consistent.